### ISSUE TITLE
On focus is triggering another search

### ISSUE THREAD

**Issue by tsuharesu** [AbdulRahmanAlHamali/flutter_typeahead#47]
Every time I click on the TypeAheadField, even if I don't type anything, `suggestionsCallback` is being triggered. I looked into `_SuggestionsListState` but couldn't figure out why `if (widget.controller.text == this._lastTextValue) return;` this is not working.

Right now I have to create some kind of check inside `suggestionsCallback` to return an empty list instead of hitting the API.

**Comment by sjmcdowall** [comment]
I have that same code in all my usages too -- something like:

```
      suggestionsCallback: (String pattern) async {
        // logger.fine('suggestionsCallback() - $pattern');
        if (pattern == _oldPattern) {
          return _recipients;
        }

        // New search string..go get the results..
        _oldPattern = pattern;
```

I didnb't even know we were trying to check it .. :)

**Comment by sjmcdowall** [comment]
Solved in 1.0.5 .. closing..
